src/main/java/com/capitalone/dashboard/collector/SonarCollectorTask.java

```diff
@@ -99,12 +116,17 @@ public void collect(SonarCollector collector) {
             for (int i = 0; i < collector.getSonarServers().size(); i++) {
 
                 String instanceUrl = collector.getSonarServers().get(i);
-                Double version = sonarClientSelector.getSonarVersion(instanceUrl);
-                String token = getToken(sonarSettings.getTokens(),i);
-
                 logBanner(instanceUrl);
+
+                Double version = sonarClientSelector.getSonarVersion(instanceUrl);
                 SonarClient sonarClient = sonarClientSelector.getSonarClient(version);
-                List<SonarProject> projects = sonarClient.getProjects(instanceUrl,token);
+
+                String username = getFromListSafely(sonarSettings.getUsernames(), i);
+                String password = getFromListSafely(sonarSettings.getPasswords(), i);
+                String token = getFromListSafely(sonarSettings.getTokens(), i);
+                sonarClient.setServerCredentials(username, password, token);
+
+                List<SonarProject> projects = sonarClient.getProjects(instanceUrl);
                 latestProjects.addAll(projects);
 
                 int projSize = ((CollectionUtils.isEmpty(projects)) ? 0 : projects.size());
```

Comment:
you can use CollectionUtils.size() which is nullsafe and checks for empty as well